### PR TITLE
Basic gibbing

### PR DESCRIPTION
**Pull request by Zumorica**

### CODE REVIEW

File: Content.Shared/GameObjects/Components/Body/SharedBodyComponent.cs [hunk 1]
```diff
@@ -697,6 +702,19 @@ public override void HandleComponentState(ComponentState? curState, ComponentSta
...
             }
         }
+
+        public virtual void Gib(bool recursive = false)
+        {
+            foreach (var (_, part) in Parts)
+            {
+                RemovePart(part);
+
+                part.Owner.RandomOffset(0.5f);
```
Comment (DrSmugleaf) [inline Content.Shared/GameObjects/Components/Body/SharedBodyComponent.cs:712]: Wouldn't this always cause a misprediction between the client and server?

File: Content.Shared/GameObjects/Components/Body/SharedBodyComponent.cs [hunk 2]
```diff
@@ -697,6 +702,19 @@ public override void HandleComponentState(ComponentState? curState, ComponentSta
                 }
             }
         }
+
+        public virtual void Gib(bool recursive = false)
```
Comment (DrSmugleaf) [inline Content.Shared/GameObjects/Components/Body/SharedBodyComponent.cs:706]: `bool gibParts` would be more descriptive here, and would also allow for `gibMechanisms` at a future date.

File: Resources/Prototypes/Body/Mechanisms/basic_human_organs.yml [hunk 3]
```diff
@@ -1,5 +1,17 @@
+- type: entity
+  id: BaseOrgan
```
Comment (DrSmugleaf) [inline Resources/Prototypes/Body/Mechanisms/basic_human_organs.yml]: Rename this to BaseOrganHuman. Bonus points if you want to separate it into BaseOrganHuman : BaseMechanism : BaseItem but I don't think that's necessary yet.

File: Content.Shared/GameObjects/Components/Body/Part/SharedBodyPartComponent.cs [hunk 4]
```diff
@@ -313,6 +314,15 @@ private void RemovedFromBody(IBody old)
         protected virtual void OnAddedToBody(IBody body) { }
 
         protected virtual void OnRemovedFromBody(IBody old) { }
+
+        public void Gib()
+        {
+            foreach (var mechanism in _mechanisms)
+            {
+                RemoveMechanism(mechanism);
+                mechanism.Owner.RandomOffset(0.25f);
```
Comment (DrSmugleaf) [inline Content.Shared/GameObjects/Components/Body/Part/SharedBodyPartComponent.cs:323]: Same misprediction problem here when called from the client.

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

File: Content.Shared/GameObjects/Components/Body/SharedBodyComponent.cs (hunk 1 above)
Comment (Zumorica) [reply Content.Shared/GameObjects/Components/Body/SharedBodyComponent.cs]: True, changed now.

File: Content.Shared/GameObjects/Components/Body/SharedBodyComponent.cs (hunk 2 above)
Comment (Zumorica) [reply Content.Shared/GameObjects/Components/Body/SharedBodyComponent.cs]: Done.

File: Resources/Prototypes/Body/Mechanisms/basic_human_organs.yml (hunk 3 above)
Comment (Zumorica) [reply Resources/Prototypes/Body/Mechanisms/basic_human_organs.yml]: I did just that

File: Content.Shared/GameObjects/Components/Body/Part/SharedBodyPartComponent.cs (hunk 4 above)
Comment (Zumorica) [reply Content.Shared/GameObjects/Components/Body/Part/SharedBodyPartComponent.cs]: Done

**Pull request merged**